components/ai_chat/resources/page/components/page_context_toggle/index.tsx

```diff
@@ -0,0 +1,64 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import Toggle from '@brave/leo/react/toggle'
+import Tooltip from '@brave/leo/react/tooltip'
+import Icon from '@brave/leo/react/icon'
+
+import styles from './style.module.scss'
+import SiteTitle from '../site_title'
+import getPageHandlerInstance from '../../api/page_handler'
+
+function PageContextToggle() {
+  const [showTooltip, setShowTooltip] = React.useState(false)
+  const [shouldSendContents, setShouldSendContents] = React.useState(true)
+
+  const handleToggleChange = (e: CustomEvent<any>) => {
+    getPageHandlerInstance().pageHandler.setShouldSendPageContents(
+      e.detail.checked
+    )
```

Comment:
This is more of a nit, but one thing concerns me here - we only show this Toggle when the associated page has SiteInfo. When we call `setShouldSendPageContents`, which this toggle is calling directly itself, we're clearing the page content and technically SiteInfo should indicate there's no page present, which would hide this very toggle as soon as it's diabled. I think that potential bug is avoided because we only re-fetch SiteInfo in f/e `initialiseForTargetTab`. It seems an easy future regression when that state gets updated whenever page association / site info changes.